test/integration/basic/test/client-navigation.js

```diff
@@ -467,7 +467,7 @@ export default (context) => {
         expect(text).toBe('10')
 
         expect(await browser.url())
-          .toBe(`http://localhost:${context.appPort}/nav/querystring/10#10`)
+          .toBe(`http://localhost:${context.appPort}/nav/querystring/10/#10`)
```

Comment:
These tests shouldn't be changed, the code in this PR is adding the wrong behavior hence why these failed.

Reply:
Reverted